pkg/distributor/distributor_test.go

```diff
@@ -1139,8 +1139,8 @@ func TestDistributor_PushQuery(t *testing.T) {
 				assert.EqualError(t, err, tc.expectedError.Error())
 
 				// Assert that downstream gRPC statuses are passed back upstream
-				_, ok := httpgrpc.HTTPResponseFromError(err)
```

Comment:
Why this change?

Reply:
This change is related to the change discussed above. 
The previous version of this test was ensuring that the error returned by `distributor.QueryStream()` was produced by using the `httpgrpc` package. This is, anyway, not true. Before error handling improvement, it was the case sometimes, but after error handling improvement it is not the case anymore. The reason why the test was not failing was because we were using `errFail` defined as `httpgrpc.Errorf(http.StatusInternalServerError, "Fail")` as the only error returned by `mockIngester`, and that made the test successful. 
Again, this could be done in a separate PR.

Reply:
I'm fine with the change in this PR, but as a general good practice I recommend to keep each PR focused on a single scope (ideally what the PR title says). As a retrospective, this test fix may have been a dedicated PR, which would have been easier to understand and in the future easier to track when we look back at changes.

_Just a feedback for the next time._